pkg/solana/client/multinode/client.go

```diff
@@ -244,7 +259,10 @@ func (m *MultiNodeClient[RPC, HEAD]) UnsubscribeAllExcept(subs ...Subscription)
 
 	for sub := range m.subs {
 		if _, keep := keepSubs[sub]; !keep {
+			// Release lock to avoid deadlock on unsubscribe
+			m.subsSliceMu.Unlock()
```

Comment:
It does not seem right. https://go.dev/doc/faq#atomic_maps
Concurrently reading/writing data from a map is not safe. I'm not 100% sure it will occur in this case, but I'd rather take a safer path.
You might be able to call Unsubscribe of underlying subscription instead of managed in the `UnsubscribeAllExcept`